File: 정수빈/LeetCode/top_interview/roman-to-integer.py
# https://leetcode.com/problems/roman-to-integer

class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        nums_tb = {
            "I" : 1,
            "IV": 4,
            "IX": 9,
            "V": 5,
            "X": 10,
            "XL": 40,
            "XC": 90,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000,
            "CM": 900,
            "CD": 400,
        }
        ans = 0
        i = 0
        while i < len(s):
            if i+1 < len(s) and s[i:i+2] in ["IV", "IX", "XL", "XC", "CM", "CD"]:
                ans += nums_tb[s[i:i+2]]
                i += 2
            else:
                ans += nums_tb[s[i:i+1]]
                i += 1
        return ans
        

File: 장세훈/LeetCode/easy/RomanToInteger.java
class RomanToInteger {

/*
    https://leetcode.com/problems/roman-to-integer/submissions/1553773756/
 */

  public int romanToInt(String s) {
    int answer = 0;
    Map<Character, Integer> map = new HashMap<>();
    map.put('I', 1);
    map.put('V', 5);
    map.put('X', 10);
    map.put('L', 50);
    map.put('C', 100);
    map.put('D', 500);
    map.put('M', 1000);

    for (int i = 0; i < s.length(); i++) {
      if (i < s.length() - 1 && map.get(s.charAt(i)) < map.get(s.charAt(i + 1))) {
        answer -= map.get(s.charAt(i));
      } else {
        answer += map.get(s.charAt(i));
      }
    }

    return answer;
  }
}